data-prepper-test-common/src/main/java/org/opensearch/dataprepper/matcher/MapEquals.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.matcher;
+
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Custom Hamcrest assertion {@link MapEquals#isEqualWithoutTimestamp(Map)} which compares two maps
+ * by ignoring {@link MapEquals#DEFAULT_TIMESTAMP_KEY_FOR_EVENT} key from maps.
+ * @since 1.3
+ */
+public class MapEquals extends TypeSafeMatcher<Map<String, Object>> {
+    private final Map<String, Object> expectedMap;
+    private final String DEFAULT_TIMESTAMP_KEY_FOR_EVENT = "@timestamp";
+
+    MapEquals(Map<String, Object> map) {
+        expectedMap = map;
+    }
+
+    @Override
+    protected boolean matchesSafely(final Map<String, Object> actualMap) {
+        Set<String> keys = new HashSet<>();
+        keys.addAll(actualMap.keySet());
+        keys.addAll(expectedMap.keySet());
+
+        for (String key : keys) {
+            Object actualValue = actualMap.get(key);
+            Object expectedValue = expectedMap.get(key);
+            if ((!key.equals(DEFAULT_TIMESTAMP_KEY_FOR_EVENT)) && (actualValue == null || expectedValue == null || !actualValue.equals(expectedValue)))
```

Comment:
This conditional will resolve to true when it should be false if both maps contain a key with a value of null.

Reply:
Should we just return false if there is a null key in any map?

Reply:
Hashmap can have null keys and values. So I think we can just use .equals method.

Reply:
Yeah this may already be what you mean, but why not just check if the timestamp key exists in both Maps. If it does, remove it and check if the Maps with it removed are equal

Reply:
Be sure to copy the Map first.

Tests may expect the timestamp after the fact.